test/unit/metrics/test_create_group_metric_set.py

```diff
@@ -0,0 +1,394 @@
+
+    result = create_group_metric_set('binary_classification', Y_true, Y_pred, Groups)
+    assert result['predictionType'] == 'binaryClassification'
+
+    assert isinstance(result['trueY'], list)
+    assert np.array_equal(result['trueY'], Y_true)
+
+    assert isinstance(result['precomputedFeatureBins'], list)
+    assert len(result['precomputedFeatureBins']) == 2
+    bin_dict0 = result['precomputedFeatureBins'][0]
+    assert isinstance(bin_dict0, dict)
+    assert np.array_equal(bin_dict0['binVector'], gr_int)
+    assert np.array_equal(bin_dict0['binLabels'], ['a', 'b'])
+    bin_dict1 = result['precomputedFeatureBins'][1]
+    assert isinstance(bin_dict1, dict)
+    assert np.array_equal(bin_dict1['binVector'], [x-4 for x in Groups[1]])
+    assert np.array_equal(bin_dict1['binLabels'], ['4', '5', '6'])
+
+    assert isinstance(result['predictedY'], list)
+    assert len(result['predictedY']) == 3
+    for i in range(3):
+        y_p = result['predictedY'][i]
+        assert isinstance(y_p, list)
+        assert np.array_equal(y_p, Y_pred[i])
+
+    assert isinstance(result['precomputedMetrics'], list)
+    assert len(result['precomputedMetrics']) == 2
+
+    # Check the first grouping (with alphabetical labels)
+    metrics_group_0 = result['precomputedMetrics'][0]
+    assert isinstance(metrics_group_0, list)
+    assert len(metrics_group_0) == 3
+    # Loop over the models
+    for i in range(3):
+        m_g0 = metrics_group_0[i]
+        assert isinstance(m_g0, dict)
+        assert len(m_g0) == 10
+        accuracy = m_g0['accuracy_score']
+        assert isinstance(accuracy, dict)
+        gmr = group_accuracy_score(Y_true, Y_pred[i], Groups[0])
+        assert gmr.overall == pytest.approx(accuracy['global'])
+        assert isinstance(accuracy['bins'], list)
+        assert len(accuracy['bins']) == 2
+        assert gmr.by_group['a'] == pytest.approx(accuracy['bins'][0])
+        assert gmr.by_group['b'] == pytest.approx(accuracy['bins'][1])
+
+    # Check the second grouping (three unique numeric labels)
+    metrics_group_1 = result['precomputedMetrics'][1]
+    assert isinstance(metrics_group_1, list)
+    assert len(metrics_group_1) == 3
+    # Loop over the models
+    for i in range(3):
+        m_g1 = metrics_group_1[i]
+        assert isinstance(m_g1, dict)
+        assert len(m_g1) == 10
+        accuracy = m_g1['accuracy_score']
+        assert isinstance(accuracy, dict)
+        gmr = group_accuracy_score(Y_true, Y_pred[i], Groups[1])
+        assert gmr.overall == pytest.approx(accuracy['global'])
+        assert isinstance(accuracy['bins'], list)
+        assert len(accuracy['bins']) == 3
+        # Use the fact that the groups are integers
+        for j in range(3):
+            assert gmr.by_group[j+4] == pytest.approx(accuracy['bins'][j])
+
+
+@pytest.mark.parametrize("transform_y_true", conversions_for_1d)
+@pytest.mark.parametrize("transform_y_pred1", conversions_for_1d)
```

Comment:
I had wanted to do the argument type testing on all the sub-lists, but it turns out that exponential growth is quite fast.